### DISCUSSION TITLE
BackgroundWorker, HTTP request and tracing

### DISCUSSION THREAD

**Discussion by AlexandreBossard** [dotnet/aspnetcore#58419] [Q&A]
I have a BackgroundWorker in my asp (net8) service that periodically polls another service through http. Contrary to a request emitted from an HttpContext (Controller or Minimal API), the HttpClient (DI'ed) request emitted from the BackgroundService does not contain the `traceparent` http header, unless I add this snippet to my Program.cs:
```
ActivitySource.AddActivityListener(new ActivityListener
{
    ShouldListenTo = _ => true,
    Sample = (ref ActivityCreationOptions<ActivityContext> _) =>
                                ActivitySamplingResult.AllData,
});
```
I have not found in the documentation a proper explanation on how to accomplish this. I have seen examples using openTelemtry, but I don't currently use it, so I would prefer not to introduce it only for this use case.
Is there a better way to do this ?

**Comment by davidfowl** [comment]
Where would the traceparent header be coming from in the background service? You're talking about an outbound http request from the background worker right?

**Reply by AlexandreBossard** [reply]
Yes, I'm talking about an outbound Http request from the BackgroundWorker. I would expect a traceparent header to be set/generated by the http client. It seems to be the behavior when doing outbound http request from an HttpContext, even if the incoming request does not contain a traceparent header. The outgoing request has one, with or without an `ActivityListener` registered.
